File: Kap13/JavaScript/Listing39_47/mygraphs/ZweiDElm.mjs
import FlaechenElm from "./FlaechenElm.mjs";

export default class ZweiDElm extends FlaechenElm {
    umfang = 0;
     
    constructor(initRandfarbe, initFuellfarbe) {
        super(initRandfarbe, initFuellfarbe);
        if (new.target === ZweiDElm) {
            throw TypeError("Error!");
        }
        console.log("ZweiDElm erzeugt.");
    }

    berechneUmfang() {
        throw TypeError("Error!");
    }
  
    getUmfang() {
        if (this.logger != null) {
            this.logger.doLog("Umfang wurde abgefragt.");
        }
        return this.umfang;
    }
  
  
    printInfo() {
      super.printInfo();
      console.log("Umfang: " + this.umfang);
    }
}	

File: Kap13/C#/Listing39_47/mygraphs/ZweiDElm.cs
namespace mygraphs {
  abstract public class ZweiDElm : FlaechenElm {
    protected double umfang = 0;
    
    public ZweiDElm(int initRandfarbe, int initFuellfarbe) : base(initRandfarbe, initFuellfarbe) {
        Console.WriteLine("ZweiDElm erzeugt.");
    }
    
    abstract protected void berechneUmfang();

    public double getUmfang() {
        if (logger != null) {
        logger.doLog("Umfang wurde abgefragt.");
        }
        return umfang;
    }


    public override void printInfo() {
        base.printInfo();
        Console.WriteLine("Umfang: " + umfang);
    }
  }	
}